### PR TITLE
Remove all usage of User.primaryEmail

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner `User.primaryEmail` is no longer updated nor used. It's still primarily read from in `UserEmailAddressRepo.getByUser` since I haven't migrated the `User` data yet (I'm on it, have a couple other PRs coming)

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala [hunk 1]
```diff
@@ -180,7 +179,7 @@ class SecureSocialUserPluginImpl @Inject() (
       val existingUserOpt = userId orElse {
         // Automatically connect accounts with existing emails
         socialUser.email.map(EmailAddress(_)).flatMap { emailAddress =>
-          emailRepo.getVerifiedOwner(emailAddress) tap {
+          emailRepo.getOwner(emailAddress) tap {
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala:182]: @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala [hunk 2]
```diff
@@ -319,18 +319,11 @@ class UserCommander @Inject() (
     segment
   }
 
-  def createUser(firstName: String, lastName: String, addrOpt: Option[EmailAddress], state: State[User]): User = {
+  def createUser(firstName: String, lastName: String, state: State[User]): User = {
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala:322]: @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala [hunk 3]
```diff
@@ -319,18 +319,11 @@ class UserCommander @Inject() (
...
     val newUser: User = db.readWrite(attempts = 3) { implicit session =>
-      val user = userRepo.save(
-        User(firstName = firstName, lastName = lastName, primaryEmail = addrOpt, state = state)
-      )
-      addrOpt.foreach { emailAddress => // TODO(ryan): this code is broken. that repo method doesn't check for active/inactive
-        if (orgInviteRepo.getByEmailAddress(emailAddress).nonEmpty) {
-          userExperimentRepo.save(UserExperiment(userId = user.id.get, experimentType = UserExperimentType.ORGANIZATION))
-        }
-      }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala:331]: @ryanpbrewster

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala [hunk 4]
```diff
@@ -94,8 +95,12 @@ class OrganizationInviteCommanderImpl @Inject() (db: Database,
...
   def convertPendingInvites(emailAddress: EmailAddress, userId: Id[User])(implicit session: RWSession): Unit = {
-    organizationInviteRepo.getByEmailAddress(emailAddress) foreach { invitation =>
+    val hasOrgInvite = organizationInviteRepo.getByEmailAddress(emailAddress).map { invitation =>
       organizationInviteRepo.save(invitation.copy(userId = Some(userId)))
+      invitation.state
+    } contains OrganizationInviteStates.ACTIVE
+    if (hasOrgInvite && !userExperimentRepo.hasExperiment(userId, UserExperimentType.ORGANIZATION)) {
+      userExperimentRepo.save(UserExperiment(userId = userId, experimentType = UserExperimentType.ORGANIZATION))
     }
   }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala:105]: @ryanpbrewster I believe this actually fixes your todo :)

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala:105]: Clever. I like it. ((What happens if that invite already has a user id on it? Is that even possible?))

**Pull request merged**